File: backend/app/schemas/options.py
"""Options flow schemas for API response."""

from datetime import date, datetime

from pydantic import BaseModel

from app.schemas.common import PaginationMeta


class OptionsActivityResponse(BaseModel):
    date: date
    total_call_volume: int | None
    total_put_volume: int | None
    total_call_oi: int | None
    total_put_oi: int | None
    put_call_ratio: float | None
    weighted_avg_iv: float | None
    atm_call_iv: float | None
    atm_put_iv: float | None
    iv_skew: float | None
    expirations_fetched: int
    data_quality: str

    model_config = {"from_attributes": True}


class PaginatedOptionsActivity(BaseModel):
    data: list[OptionsActivityResponse]
    meta: PaginationMeta


class CboePutCallResponse(BaseModel):
    date: date
    total_pc: float | None
    equity_pc: float | None
    index_pc: float | None

    model_config = {"from_attributes": True}

File: frontend/src/types/options.ts
export interface OptionsActivity {
  date: string;
  total_call_volume: number | null;
  total_put_volume: number | null;
  total_call_oi: number | null;
  total_put_oi: number | null;
  put_call_ratio: number | null;
  weighted_avg_iv: number | null;
  atm_call_iv: number | null;
  atm_put_iv: number | null;
  iv_skew: number | null;
  expirations_fetched: number;
  data_quality: "full" | "partial" | "stale";
}

export interface CboePutCallRatio {
  date: string;
  total_pc: number | null;
  equity_pc: number | null;
  index_pc: number | null;
}
